lib/puma/client.rb

```diff
@@ -166,7 +168,10 @@ def reset(fast_check=true)
       @peerip = nil if @remote_addr_header
       @in_last_chunk = false
       @http_content_length_limit_exceeded = false
+    end
 
+    # only used with pipelined (back-to-back) requests
+    def fast_try_to_finish
```

Comment:
Request: The consumer of this method relies on the return value to be truthy/falsey. We should document what that is supposed to mean here.

> Assumes that this method is only called on a `keepalive` connection where one connection can contain multiple requests
> ...
> A truthy return value indicates that the client contains an additional request and that request is fully buffered and ready to be handled in the client pool
> A falsey return value indicates that

Reply:
Request: Short I would like to consider a more meaningful name

Long:
The naming "fast" is historically consistent, but possibly misleading. I'm looking at the body of `try_to_finish` and it looks like it's doing a lot less. It doesn't read from the socket, it only parses what it already has (if there is a buffered/additional request).

Maybe a name involving pipelines? One option: `try_pipeline_finish`. Or maybe something that indicates it doesn't read from IO like `try_buffer_finish` or `non_read_try_finish`? I'm open to suggestions.